go/common/viewingkey/viewing_key_messages.go

```diff
@@ -0,0 +1,258 @@
+	}
+
+	message := map[string]interface{}{
+		EIP712EncryptionToken: encryptionToken,
+	}
+
+	types := apitypes.Types{
+		EIP712Domain: {
+			{Name: EIP712DomainName, Type: "string"},
+			{Name: EIP712DomainVersion, Type: "string"},
+			{Name: EIP712DomainChainID, Type: "uint256"},
+		},
+		EIP712Type: {
+			{Name: EIP712EncryptionToken, Type: "address"},
+		},
+	}
+
+	newTypeElement := apitypes.TypedData{
+		Types:       types,
+		PrimaryType: EIP712Type,
+		Domain:      domain,
+		Message:     message,
+	}
+
+	rawData, err := getBytesFromTypedData(newTypeElement)
+	if err != nil {
+		return nil, err
+	}
+
+	// add the JSON message to the list of messages
+	jsonData, err := json.Marshal(newTypeElement)
+	if err != nil {
+		return nil, err
+	}
+
+	if hash {
+		return crypto.Keccak256(rawData), nil
+	}
+	return jsonData, nil
+}
+
+// GenerateMessage generates a message for the given encryptionToken, chainID, version and signatureType
+// hashed parameter is used to determine if the returned message should be hashed or returned in plain text
+func GenerateMessage(encryptionToken string, chainID int64, version int, signatureType SignatureType, hashed bool) ([]byte, error) {
+	generator, exists := messageGenerators[signatureType]
+	if !exists {
+		return nil, fmt.Errorf("unsupported signature type")
+	}
+	return generator.generateMessage(encryptionToken, chainID, version, hashed)
+}
+
+// GenerateSignMessage creates the message to be signed
+// vkPubKey is expected to be a []byte("0x....") to create the signing message
+// todo (@ziga) Remove this method once old WE endpoints are removed
+func GenerateSignMessage(vkPubKey []byte) string {
+	return SignedMsgPrefix + hex.EncodeToString(vkPubKey)
+}
+
+// getBytesFromTypedData creates EIP-712 compliant hash from typedData.
+// It involves hashing the message with its structure, hashing domain separator,
+// and then encoding both hashes with specific EIP-712 bytes to construct the final message format.
+func getBytesFromTypedData(typedData apitypes.TypedData) ([]byte, error) {
+	typedDataHash, err := typedData.HashStruct(typedData.PrimaryType, typedData.Message)
+	if err != nil {
+		return nil, err
+	}
+	// Create the domain separator hash for EIP-712 message context
+	domainSeparator, err := typedData.HashStruct(EIP712Domain, typedData.Domain.Map())
+	if err != nil {
+		return nil, err
+	}
+	// Prefix domain and message hashes with EIP-712 version and encoding bytes
+	rawData := append([]byte("\x19\x01"), append(domainSeparator, typedDataHash...)...)
+	return rawData, nil
+}
+
+// GenerateViewingKeyForWallet takes an account wallet, generates a viewing key and signs the key with the acc's private key
+// uses the same method of signature handling as Metamask/geth
+// TODO @Ziga - update this method to use the new EIP-712 signature format / personal sign after the removal of the legacy format
+func GenerateViewingKeyForWallet(wal wallet.Wallet) (*ViewingKey, error) {
```

Comment:
this method and its dependents don't feel like they belong in this file.
I think they would better live in "viewing_key"

Reply:
moved those functions back